src/Stores/Promises/selectors.js

```diff
@@ -23,9 +23,22 @@ export const promiseSubjectsSelector: (
       const links = subjectsPromises
         ? subjectsPromises.filter(sP => sP.promise === promise.id)
         : [];
-      const subs = links.map(
-        link => (subjects ? subjects.find(s => link.subject === s.id) : [])
-      );
+      const subs = links.map(link => {
+        if (!subjects) {
+          // TODO trigger loading of subjects instead of throwing error
+          throw new Error('trying to find subjects when they are not loaded');
+        }
+        const subject = {
+          ...subjects.find(s => link.subject === s.id),
+          subjectPromiseId: link.id,
+        };
+
+        if (!subject) {
+          throw new Error('subject promises link without subject');
```

Comment:
This ever going to happen? Wont subjects always be truthy?

Reply:
Not if they aren't loaded.

Once Sentry has been setup, it'll catch these errors that shouldn't happen.

But it's better to throw an error rather than failing silently

Reply:
http://pragmatictips.com/33